core/base/mtx_io.cpp

```diff
@@ -693,7 +693,9 @@ class mtx_io {
             "    <LAYOUT-TYPE>     is one of: coordinate, array\n"
             "    <VALUE-TYPE>      is one of: real, integer, complex, pattern\n"
             "    <LAYOUT-MODIFIER> is one of: general, symmetric, "
-            "skew-symmetric, hermitian\n");
+            "skew-symmetric, hermitian\n"
+            "Header instead: " +
```

Comment:
I think this could be made clearer, e.g. something like "Found the following header line instead:"